components/LineText.js

```diff
@@ -7,21 +7,35 @@ import React from "react";
 import { StyleSheet, View, Text } from "react-native";
 import Color from "../constants/Colors";
 
-const LineHeader = props => {
+const LineText = props => {
+  const { textStyle, lineWeight, lineColor } = props;
+
+  let addedLineProps = {};
+  if (lineWeight) {
+    addedLineProps["borderWidth"] = lineWeight;
+  }
+  if (lineColor) {
+    addedLineProps["borderColor"] = lineColor;
+  }
   return (
-    <View>
-      <Text style={{ ...styles.text, ...props.style }}>{props.children}</Text>
-      <View style={styles.lineStyle} />
+    <View style={{ ...styles.comp, ...props.style }}>
+      <Text style={{ ...styles.text, ...textStyle }}>{props.children}</Text>
+      <View style={{ ...styles.lineStyle, ...addedLineProps }} />
```

Comment:
Why not add here: ...props.lineStyle and add this style from the outside of the component?
All the "if" statement and the new object "addedLineProps" could be deleted.